From 7b5a2bf1196e8c3e592aa7f02dd9c57576ce2fd7 Mon Sep 17 00:00:00 2001
From: Henry Wu <Henry_Wu@quantatw.com>
Date: Mon, 27 Jul 2026 17:59:48 +0800
Subject: [PATCH 25/25] oem-meta: santabarbara: update event logs from rainbow
 MMC

Add new logs from Rainbow ASIC via MMC

Test on Santabarbara platform
```
    "13": {
        "additional_data": {
            "DEVICE": "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow_0/irq_tmp75_1_alert_r_n",
            "FAILURE_DATA": "RAINBOW_IRQ_TMP75_1_ALERT_R_N - [02 03]",
            "_CODE_FILE": "/usr/src/debug/pldm/1.0+git/oem/meta/libpldmresponder/file_io_type_event_log.cpp",
            "_CODE_FUNC": "void pldm::responder::oem_meta::record::commit(const std::string&, pldm::responder::oem_meta::EventAssert, const std::string&) [with const char* TypeLabel = (& label); AssertType = sdbusplus::error::xyz::openbmc_project::state::Thermal::DeviceOverOperatingTemperatureFault; DeassertType = sdbusplus::event::xyz::openbmc_project::state::Thermal::DeviceOperatingNormalTemperature; std::string = std::__cxx11::basic_string<char>]",
            "_CODE_LINE": "201",
            "_PID": "1752"
        },
        "event_id": "",
        "message": "xyz.openbmc_project.State.Thermal.DeviceOverOperatingTemperatureFault",
        "redfish": {
            "args": [
                "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow_0/irq_tmp75_1_alert_r_n"
            ],
            "id": "OpenBMC_StateThermal.DeviceOverOperatingTemperatureFault",
            "message": "Device /xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow_0/irq_tmp75_1_alert_r_n is significantly over safe operating temperature and may have been powered off."
        },
        "resolution": "",
        "resolved": false,
        "severity": "xyz.openbmc_project.Logging.Entry.Level.Critical",
        "timestamp": "2026-07-23T12:50:53.825000000Z",
        "updated_timestamp": "2026-07-23T12:50:53.825000000Z"
    }
}
```

Gerrit Link:
https://gerrit.openbmc.org/c/openbmc/pldm/+/92664

Signed-off-by: Henry Wu <Henry_Wu@quantatw.com>
---
 .../file_io_type_event_log.cpp                | 122 +++++++++++++++++-
 1 file changed, 121 insertions(+), 1 deletion(-)

diff --git a/oem/meta/libpldmresponder/file_io_type_event_log.cpp b/oem/meta/libpldmresponder/file_io_type_event_log.cpp
index 9f4d4ea..bcec9a3 100644
--- a/oem/meta/libpldmresponder/file_io_type_event_log.cpp
+++ b/oem/meta/libpldmresponder/file_io_type_event_log.cpp
@@ -583,7 +583,7 @@ static const std::map<uint8_t, FaultDesc> rainbowFaultsList = {
       "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p12v_ubc_pwrgd",
       "VOLTAGE_REGULATOR", record::voltageRegulatorFault}},
     {0x26,
-     {"RAINBOW_P0V75_AVDD_HCSL ",
+     {"RAINBOW_P0V75_AVDD_HCSL",
       "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_avdd_hcsl",
       "VOLTAGE_REGULATOR", record::voltageRegulatorFault}},
     {0x27,
@@ -654,6 +654,126 @@ static const std::map<uint8_t, FaultDesc> rainbowFaultsList = {
      {"RAINBOW_CLK_BUF2_100M_LOSB_PLD",
       "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/clk_buf2_100m_losb_pld",
       "DEVICE", record::powerRailFault}},
+    {0x60,
+     {"RAINBOW_ASIC_MODULE_ERROR",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/asic_module_error",
+      "DEVICE", record::powerRailFault}},
+    {0x61,
+     {"RAINBOW_P0V9_OWL_E_TRVDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v9_owl_e_trvdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x62,
+     {"RAINBOW_P0V75_OWL_E_TRVDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_owl_e_trvdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x63,
+     {"RAINBOW_P0V75_OWL_E_VDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_owl_e_vdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x64,
+     {"RAINBOW_P0V9_OWL_W_TRVDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v9_owl_w_trvdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x65,
+     {"RAINBOW_P0V75_OWL_W_TRVDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_owl_w_trvdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x66,
+     {"RAINBOW_P0V75_OWL_W_VDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_owl_w_vdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x67,
+     {"RAINBOW_P0V75_MAX_M_VDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_max_m_vdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x68,
+     {"RAINBOW_P0V75_MAX_N_VDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_max_n_vdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x69,
+     {"RAINBOW_P0V75_MAX_S_VDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_max_s_vdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x6A,
+     {"RAINBOW_P0V8_HAMSA_AVDD_PCIE_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v8_hamsa_avdd_pcie_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x6B,
+     {"RAINBOW_P1V2_HAMSA_VDDHRXTX_PCIE_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p1v2_hamsa_vddhrxtx_pcie_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x6C,
+     {"RAINBOW_P0V85_HAMSA_VDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v85_hamsa_vdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x6D,
+     {"RAINBOW_P0V75_VDDPHY_HBM0246_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_vddphy_hbm0246_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x6E,
+     {"RAINBOW_P0V4_VDDQL_HBM0246_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v4_vddql_hbm0246_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x6F,
+     {"RAINBOW_P1V1_VDDQC_HBM0246_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p1v1_vddqc_hbm0246_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x70,
+     {"RAINBOW_P1V8_VPP_HBM0246_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p1v8_vpp_hbm0246_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x71,
+     {"RAINBOW_P0V75_VDDPHY_HBM1357_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v75_vddphy_hbm1357_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x72,
+     {"RAINBOW_P0V4_VDDQL_HBM1357_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p0v4_vddql_hbm1357_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x73,
+     {"RAINBOW_P1V1_VDDQC_HBM1357_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p1v1_vddqc_hbm1357_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x74,
+     {"RAINBOW_P1V8_VPP_HBM1357_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/p1v8_vpp_hbm1357_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x75,
+     {"RAINBOW_TEMP_MON2_OVERT_N",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/temp_mon2_overt_n",
+      "DEVICE", record::thermalFault}},
+    {0x76,
+     {"RAINBOW_TEMP_MON3_OVERT_N",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/temp_mon3_overt_n",
+      "DEVICE", record::thermalFault}},
+    {0x77,
+     {"RAINBOW_TEMP_MON4_OVERT_N",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/temp_mon4_overt_n",
+      "DEVICE", record::thermalFault}},
+    {0x78,
+     {"RAINBOW_TEMP_MON5_OVERT_N",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/temp_mon5_overt_n",
+      "DEVICE", record::thermalFault}},
+    {0x79,
+     {"RAINBOW_IRQ_TMP75_3_ALERT_R_N",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/irq_tmp75_3_alert_r_n",
+      "DEVICE", record::thermalFault}},
+    {0x7A,
+     {"RAINBOW_IRQ_TMP75_2_ALERT_R_N",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/irq_tmp75_2_alert_r_n",
+      "DEVICE", record::thermalFault}},
+    {0x7B,
+     {"RAINBOW_IRQ_TMP75_1_ALERT_R_N",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/irq_tmp75_1_alert_r_n",
+      "DEVICE", record::thermalFault}},
+    {0x7C,
+     {"RAINBOW_ASIC_P0V85_MEDHA0_VDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/asic_p0v85_medha0_vdd_ot_warning",
+      "DEVICE", record::thermalFault}},
+    {0x7D,
+     {"RAINBOW_ASIC_P0V85_MEDHA1_VDD_OT_WARNING",
+      "/xyz/openbmc_project/inventory/system/board/Santabarbara_Rainbow/asic_p0v85_medha1_vdd_ot_warning",
+      "DEVICE", record::thermalFault}},
 };
 
 struct EventContext
-- 
2.47.3

